osu.Game/Rulesets/Mods/ModPitchShift.cs

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Audio;
+using osu.Framework.Audio.Track;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Configuration;
+using osu.Game.Rulesets.UI;
+
+namespace osu.Game.Rulesets.Mods
+{
+    public class ModPitchShift : Mod, IUpdatableByPlayfield, IApplicableToTrack
+    {
+        public override string Name => "Pitch Shift";
+        public override string Acronym => "PS";
+        public override IconUsage? Icon => FontAwesome.Solid.WaveSquare;
+        public override ModType Type => ModType.Conversion;
+        public override string Description => "Raise or lower the track's pitch.";
+        public override double ScoreMultiplier => 1;
+
+        [SettingSource("Pitch shift", "Raise or lower the track's pitch")]
+        public BindableNumber<double> PitchChange { get; } = new BindableDouble
+        {
+            MinValue = 0.5,
+            MaxValue = 2.0,
+            Default = 1,
+            Value = 1,
+            Precision = 0.01,
+        };
+
+        [SettingSource("Match tempo", "Match pitch with the tempo")]
+        public BindableBool MatchTempo { get; } = new BindableBool();
+
+        private readonly BindableNumber<double> tempoAdjust = new BindableDouble(1);
+        private readonly BindableNumber<double> freqAdjust = new BindableDouble(1);
+
+        private ITrack track;
+
+        public ModPitchShift()
+        {
+            PitchChange.BindValueChanged(val =>
+                setPitch(val.NewValue), true);
+
+            MatchTempo.BindValueChanged(matchTempoChanged);
+        }
+
+        public void ApplyToTrack(ITrack track)
+        {
+            this.track = track;
+
+            track.AddAdjustment(AdjustableProperty.Tempo, tempoAdjust);
+            track.AddAdjustment(AdjustableProperty.Frequency, freqAdjust);
+        }
+
+        public void Update(Playfield playfield)
+        {
+            if (MatchTempo.Value)
```

Comment:
the bindable isn't being updated any more, so i believe it will report the wrong setting value on tooltips etc.

Reply:
Not quite sure if this is what you meant but is this not the correct behaviour?
MatchTempo off
![image](https://user-images.githubusercontent.com/7267697/108785087-a4fc9f80-7568-11eb-87a4-0d1c077c47d2.png)

MatchTempo on
![image](https://user-images.githubusercontent.com/7267697/108785141-ba71c980-7568-11eb-8537-8bb815b0c41c.png)